src/python/pants/core/util_rules/archive.py

```diff
@@ -19,33 +20,58 @@
 from pants.util.logging import LogLevel
 
 
-@dataclass(frozen=True)
-class MaybeExtractable:
-    """A newtype to work around rule graph resolution issues.
+# Note that updating this will impact the `archive` target defined in `core/target_types.py`.
+class ArchiveFormat(Enum):
+    TAR = "tar"
+    TGZ = "tgz"
+    TBZ2 = "tbz2"
+    TXZ = "txz"
```

Comment:
Lmk if we should be using the long form, like `tar.gz`.

Reply:
I've never seen `.tgz` before today but it seems like they're exactly the same. IMO I'd stick with `.tar.gz` due to being more "common" but if they behave the same way it's more "cosmetics" ?

Reply:
It's easy to support both probably?

Reply:
`.tar.gz` is more common, I think it's sufficient to just support that for now.

Reply:
I'll use the long form. Thanks!